modules/checkout/tests/src/Functional/CheckoutFlowTest.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Drupal\Tests\commerce_checkout\Functional;
+
+use Drupal\Tests\commerce\Functional\CommerceBrowserTestBase;
+use Drupal\commerce_checkout\Entity\CheckoutFlow;
+
+/**
+ * Tests the checkout flow UI.
+ *
+ * @group commerce
+ */
+class CheckoutFlowTest extends CommerceBrowserTestBase {
+
+  /**
+   * Modules to enable.
+   *
+   * @var array
+   */
+  public static $modules = [
+    'commerce_checkout',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getAdministratorPermissions() {
+    return array_merge([
+      'administer checkout flows',
+    ], parent::getAdministratorPermissions());
+  }
+
+  /**
+   * Tests creating a checkout flow.
+   */
+  public function testCheckoutFlowCreation() {
+    $this->drupalGet('admin/commerce/config/checkout-flows');
+    $this->getSession()->getPage()->clickLink('Add checkout flow');
+    $edit = [
+      'label' => 'Test checkout flow',
+      'id' => 'test_checkout_flow',
+      'plugin' => 'multistep_default',
+    ];
+    $this->submitForm($edit, 'Save');
+
+    // Test UI shows the correct message.
+    $this->assertSession()->pageTextContains(t("Saved the @name checkout flow.", ['@name' => $edit['label']]));
+
+    // Test entity creation.
+    $checkout_flow = CheckoutFlow::load('test_checkout_flow');
+    $this->assertEquals('Test checkout flow', $checkout_flow->label());
+    $this->assertEquals('multistep_default', $checkout_flow->getPluginId());
+  }
+
+  /**
+   * Tests editing a checkout flow.
+   */
+  public function testCheckoutFlowEditing() {
+    // @todo Test default flow is used when checking out.
```

Comment:
We can remove this todo, such a test case would belong to a different test, this only only concerns itself with the editing of the checkout flow entity.